src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/OnAutoInsertHandler.cs

```diff
@@ -73,24 +84,34 @@ public async Task<DocumentOnAutoInsertResponseItem> HandleRequestAsync(DocumentO
                 return null;
             }
 
+            _logger.LogInformation($"Starting request for {request.TextDocument.Uri}, with trigger character {request.Character}.");
+
             if (!_documentManager.TryGetDocument(request.TextDocument.Uri, out var documentSnapshot))
             {
                 return null;
             }
 
             var projectionResult = await _projectionProvider.GetProjectionAsync(documentSnapshot, request.Position, cancellationToken).ConfigureAwait(false);
-            if (projectionResult == null || projectionResult.LanguageKind == RazorLanguageKind.Razor)
+            if (projectionResult == null)
             {
+                _logger.LogWarning($"Failed to find document {request.TextDocument.Uri}.");
+                return null;
+            }
+            else if (projectionResult.LanguageKind == RazorLanguageKind.Razor)
+            {
+                _logger.LogInformation("OnAutoInsert not supported in Razor context.");
                 return null;
             }
             else if (projectionResult.LanguageKind == RazorLanguageKind.Html &&
                 !HTMLAllowedTriggerCharacters.Contains(request.Character, StringComparer.Ordinal))
             {
+                _logger.LogInformation("Invalid HTML trigger char.");
                 return null;
             }
             else if (projectionResult.LanguageKind == RazorLanguageKind.CSharp &&
                 !CSharpAllowedTriggerCharacters.Contains(request.Character, StringComparer.Ordinal))
             {
+                _logger.LogInformation("Invalid C# trigger char.");
```

Comment:
Invalid may be a little strong. Might be better to say "inapplicable C# trigger character, no-oping"

Reply:
Maybe I'm being pedantic but I feel like adding the "no-oping" is weird. I see logs as "I did X"  as opposed to "I have done X and will now do the following control-flow".